src/microbe_stage/editor/upgrades/ChemoreceptorUpgradeGUI.cs

```diff
@@ -121,4 +187,39 @@ protected override void Dispose(bool disposing)
 
         base.Dispose(disposing);
     }
+
+    /// <summary>
+    ///   Sets the GUI up to reflect an existing configuration
+    /// </summary>
+    private void LoadConfiguration(ChemoreceptorUpgrades configuration,
+        List<Compound> shownCompoundChoices, List<Species> shownSpeciesChoices)
+    {
+        if (configuration.TargetCompound != null)
+        {
+            TypeChanged(0);
+            targetTypes.Selected = 0;
+            compounds.Selected = shownCompoundChoices.FindIndex(c => c == configuration.TargetCompound);
+        }
+        else if (configuration.TargetSpecies != null)
+        {
+            TypeChanged(1);
+            targetTypes.Selected = 1;
+            species.Selected = shownSpeciesChoices.FindIndex(c => c == configuration.TargetSpecies);
```

Comment:
Does this make a sensible selection when the species is extinct and is no longer found? My hunch is that it does, but I just want to ask to confirm.

Reply:
Yes, since the configuration saves the species/compound itself this should be robust against changes in the world. On entering the editor if the species is no longer found it sets the `OptionButton` to unselected. If it stays unselected it is set to the first species in the list on `ApplyChanges`. If the editor stage is skipped it is possible that an extinct species stays in the configuration. Then the component simply does nothing until edited again.

Reply:
That sounds about what I expected would happen, which is a fine approach.